### PR TITLE
Tokenomic: numeric, reward split & slash to treasury, edge cases

### LINKED ISSUE #259: Tokenomic implementation in GK
Blocked by Gatekeeper implementation (#255)

- [x] Tokenomic design
- [x] Mining pallet
- [x] pRuntime heartbeat handling
- [x] Tokenomic implementation
- [x] Unit tests & adjust parameter
- [ ] Numeric end to end test

### PR DESCRIPTION
**Pull request by h4x3rotab**
Fixes #259, #342 

Please don't merge before #383

### CODE REVIEW

File: standalone/pruntime/enclave/src/system/gk.rs [hunk 1]
```diff
@@ -1302,46 +1317,67 @@ pub mod tests {
...
                 r.gk.process_messages(block);
             });
         }
-        info!("slashed v = {}", r.get_worker(0).tokenomic.v);
+        assert!(r.get_worker(0).unresponsive);
+        let report = r.gk.egress.drain_mining_info_update_event();
+        assert_eq!(report[0].offline, vec![r.workers[0].clone()]);
+        assert_eq!(r.get_worker(0).tokenomic.v, fp!(2997.0260877851113935014));
     }
 }
```
Comment (kvinwang) [inline standalone/pruntime/enclave/src/system/gk.rs:1383]: Can we cover more numeric cases?

Comment (h4x3rotab) [reply standalone/pruntime/enclave/src/system/gk.rs:1385]: Left a TODO. Let's merge this one first

File: pallets/phala/src/mining.rs [hunk 2]
```diff
@@ -48,14 +48,25 @@ pub mod pallet {
 			matches!(self, MinerState::Ready | MinerState::MiningCoolingDown)
 		}
 		fn can_settle(&self) -> bool {
+			// TODO(hangyin):
+			//
+			// We don't allow a settlement in MiningCoolingDown or Ready. After a mienr is stopped,
+			// it's relesed immediately and the slash is pre-settled (to make sure the force
```
Comment (kvinwang) [inline pallets/phala/src/mining.rs:54]: typo: 
mienr -> miner
relesed -> released

Comment (h4x3rotab) [reply pallets/phala/src/mining.rs]: Fixed. Thanks

**Pull request merged**